kernel/lib/time.cpp

```diff
@@ -0,0 +1,104 @@
+/**
+ * @file time.hpp
+ * @author Michel (JMallone) Gomes (michels@utfpr.edu.br)
+ * @brief Time lib code 
+ * @version 0.1
+ * @date 2021-07-27
+ *
+ * @copyright Copyright the Panix Contributors (c) 2021
+ *
+ * References:
+ *         https://github.com/sidsingh78/EPOCH-to-time-date-converter/blob/master/epoch_conv.c
+ *         https://www.oryx-embedded.com/doc/date__time_8c_source.html
+ *         https://stackoverflow.com/questions/11188621/how-can-i-convert-seconds-since-the-epoch-to-hours-minutes-seconds-in-java/11197532#11197532
+ */
+#include <dev/rtc/rtc.hpp>
+#include <dev/tty/tty.hpp>
+#include <lib/stdio.hpp>
+#include <lib/time.hpp>
+namespace Time {
+TimeDescriptor::TimeDescriptor()
+{
+    toDate();
+}
+
+TimeDescriptor::TimeDescriptor(int sec, int min, int hr, int d, int m, int y)
+    : _seconds(sec)
+    , _minutes(min)
+    , _hour(hr)
+    , _day(d)
+    , _month(m)
+    , _year(y)
+{
+    // Initialize nothing.
+}
+TimeDescriptor::~TimeDescriptor()
+{
+    // Nothing to deconstruct
+}
+
+void TimeDescriptor::converterEpochToDate(uint64_t epoch)
+{
+    setSeconds(epoch % 60);  // Set the seconds in epoch and set
+    epoch /= 60;             // 'Whithdraw' the seconds 
+    setMinutes(epoch % 60);  // Set the Minutes
+    epoch /= 60;             // 'Whithdraw' the Minutes
+    setHour(epoch % 24);     // Set the Hour 
+    epoch /= 24;             // 'Whithdraw' the Hour
+
+    int daysSinceJan1st[2][13] = {
+        { 0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334 }, // 365 days, non-leap
+        { 0, 31, 60, 91, 121, 152, 182, 213, 244, 274, 305, 335 }  // 366 days, leap
+    };
+
+    int epoch_in_jnd = 2440588;                     // 1/1/1970 converted in days since year 0 on julian calendar
+    int gregorian_in_jnd = 2299161;                 // 15/10/1582 begin of gregorian calendar
+    int sum_jnd = epoch + epoch_in_jnd;             // how many days happens since the 1/1/1970
+    int count_days = sum_jnd - gregorian_in_jnd;    // how many days happens since 15/10/1582
+    count_days -= 78;                               // minus 78, for easy do math, because its stated in 1/1/1583
+    int years_tmp = count_days / 365.2425;          // one year in gregorian calendar is 365.2425 days 
+    int year = years_tmp + 1583;                    //  (my_year - 1970) + 1583 = year in gregorian alendar
+
+    float month_tmp = count_days / 365.2425;        //      That is a MOD Function with
```

Comment:
So fun fact, we can't actually use `float` types in the kernel.
This code seems *way* more complicated than what was used before, so what I might suggest doing is using what we had before and creating `#define`s for some of the larger numbers like `2440588`, `2299161`, etc.
I'm just very nervous about doing so much more work when the previous code seemed to be simpler...